spring-xd-module/src/main/java/org/springframework/xd/module/ModuleDescriptor.java

```diff
@@ -575,6 +575,9 @@ public Builder setModuleDefinition(ModuleDefinition moduleDefinition) {
 		}
 
 		public ModuleDefinition getModuleDefinition() {
+			if (moduleDefinition == null) {
+				moduleDefinition = ModuleDefinitions.dummy(moduleName, type);
```

Comment:
Please don't use `ModuleDefinitions.dummy()`

If there really is a migration issue, then recreate the correct definitions (or tell the users to manually redeploy), but don't use dummy resources  in production code